processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -462,6 +482,50 @@ else if ( propertyEntries.size() == 1 ) {
             }
         }
 
+        private Assignment getOrcreateWrapperByValueCheckStrategy(Assignment assignment, boolean isUpdate) {
+
+            if ( valueSetCheckStrategy == null ) {
+                return createNullCheckWrapper( assignment, isUpdate );
+            }
+
+            switch ( valueSetCheckStrategy ) {
+                case IS_NULL_INLINE:
+                    return createNullCheckWrapper( assignment, isUpdate );
+
+                case CUSTOM:
+                    String hasMethod = this.getSourceHasMethod();
+
+                    if ( hasMethod == null ) {
+                        throw new IllegalArgumentException( sourceReference.getParameter().getName()
```

Comment:
Better not throw an exception from the AP environment. Instead we should create actual compilation errors which are gracefully handled by the processing infrastructure. That way the errors are also linked to the line number of the flawed mapping method.

You can find examples in this class, e.g. like so:

    ctx.getMessager().printMessage( method.getExecutable(),
        Message.SOME_KEY,
        targetPropertyName
    );

Instead of "hasX()",  the actual name of the value presence check method would be nice to read:

> Using custom source value presence checking strategy, but no method 'boolean hasMyProperty()` was found.